packages/validator/src/util/externalSignerClient.ts

```diff
@@ -160,17 +162,68 @@ export async function externalSignerUpCheck(remoteUrl: string): Promise<boolean>
     headers: {"Content-Type": "application/json"},
   });
 
-  const data = await handlerExternalSignerResponse<{status: string}>(res);
+  const data = await handlerExternalSignerUpCheckResponse(res);
   return data.status === "OK";
 }
 
-async function handlerExternalSignerResponse<T>(res: Response): Promise<T> {
+async function handlerExternalSignerGetKeysResponse<T>(res: Response, logger?: Logger): Promise<string[]> {
   if (!res.ok) {
     const errBody = await res.text();
+    logger?.debug("External signer GetKeys error", {body: errBody});
     throw Error(`${errBody}`);
   }
 
-  return JSON.parse(await res.text()) as T;
+  const resBody = await res.text();
+  const resContentType = res.headers.get("Content-Type");
```

Comment:
This is not robust enough, would suggest something like

```ts
res.headers.get("Content-Type").split(";", 1)[0].trim().toLowerCase()
```

Reply:
Fixed